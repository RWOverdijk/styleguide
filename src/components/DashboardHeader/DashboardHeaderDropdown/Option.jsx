import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../../Icon';

const Option = ({ icon, caption, subcaption, handleClick }) => (
  <div className="Van-DashboardHeaderDropdownEntry" onClick={handleClick}>
    {icon !== null && (
      <div className="Van-DashboardHeaderDropdownEntry-iconWrapper">
        <Icon source={icon} />
      </div>
    )}

    <div className="Van-DashboardHeaderDropdownEntry-option">
      <div className="Van-DashboardHeaderDropdownEntry-caption">{caption}</div>

      {subcaption && (
        <div className="Van-DashboardHeaderDropdownEntry-subcaption">{subcaption}</div>
      )}
    </div>
  </div>
);

Option.propTypes = {
  icon: PropTypes.string,
  caption: PropTypes.node.isRequired,
  subcaption: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
};

Option.defaultProps = {
  icon: null,
  subcaption: null
};

export default Option;
