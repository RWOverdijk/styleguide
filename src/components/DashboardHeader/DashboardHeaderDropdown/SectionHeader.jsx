import React from 'react';
import PropTypes from 'prop-types';

const SectionHeader = ({ title }) => (
  <div className="Van-DashboardHeaderDropdown-sectionHeader">{title}</div>
);

SectionHeader.propTypes = {
  title: PropTypes.node.isRequired
};

export default SectionHeader;
