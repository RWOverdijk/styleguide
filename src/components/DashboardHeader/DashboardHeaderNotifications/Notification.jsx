import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../../Icon';
import iconClose from '../../../assets/icons/fa/close.svg';

const Notification = ({ close, title, children }) => (
  <div className="Van-DashboardHeaderNotifications-notification">
    {close && (
      <div className="Van-DashboardHeaderNotifications-notificationClose" onClick={close}>
        <Icon source={iconClose} />
      </div>
    )}

    {title && <p className="Van-DashboardHeaderNotifications-title">{title}</p>}

    {children}
  </div>
);

Notification.propTypes = {
  close: PropTypes.func,
  children: PropTypes.node.isRequired,
  title: PropTypes.node.isRequired,
};

Notification.defaultProps = {
  close: null,
};

export default Notification;
