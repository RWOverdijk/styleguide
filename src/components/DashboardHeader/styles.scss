@import '../../css/variables';

.Van-DashboardHeader {
  background-color: white;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 0 -15px;
  padding: 0 10px;

  @media (min-width: $container-sm) {
    padding: 0 15px;
  }

  &-logoWrapper {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-content: flex-start;
    margin: 15px 0;
  }

  &-logoText {
    margin-right: 5px;
    font-size: 23.1px;
    font-weight: lighter;

    @media (min-width: $container-sm) {
      font-size: 25px;
    }
  }

  &-logo {
    margin-top: 5px;
    width: 109px;

    @media (min-width: $container-sm) {
      width: 135px;
    }
  }

  &-addons {
    display: flex;
    flex-direction: row;
    align-items: center;
  }
}

.Van-DashboardHeaderDropdown {
  padding: 0 10px;
  height: 100%;
  display: flex;
  position: relative;

  @media (min-width: $container-sm) {
    padding: 0 15px;
  }

  &:focus {
    outline: 0;
  }

  &--active {
    background-color: $color-grey-bright;
  }

  &-iconWrapper {
    margin-right: 15px;

    svg {
      fill: $color-grey-nobel;
    }
  }

  &-captionWrapper {
    display: flex;
    align-items: center;
    cursor: pointer;

    &:focus {
      outline: 0;
    }
  }

  &-caption {
    text-align: left;
  }

  &-title {
    font-size: 18px;
    font-weight: 600;
  }

  &-subtitle {
    color: $color-grey;
    font-size: 14px;
    font-weight: lighter;
    margin-top: 2px;
  }

  &-toggle {
    margin-left: 30px;

    // svg {
    //   fill: $color-grey;
    // }
  }

  &-optionWrapper {
    position: absolute;
    top: 100%;
    right: 0;
    left: 0;
    background-color: white;
    z-index: 20;

    > * {
      border: 1px solid $color-grey-bright;

      &:not(:first-child) {
        border-top: 0;
      }
    }
  }

  &-sectionHeader {
    padding: 15px;
    font-size: 18px;
    font-weight: 600;
    color: $color-grey;
  }

  @media (max-width: $container-sm) {
    &-iconWrapper {
      margin: 0;
    }

    &-optionWrapper {
      left: initial;
      width: 216px;
    }

    &-caption,
    &-toggle {
      display: none;
    }
  }
}

.Van-DashboardHeaderDropdownEntry {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  padding: 15px;
  cursor: pointer;
  z-index: 20;

  &-iconWrapper {
    margin-right: 15px;

    svg {
      fill: $color-grey-nobel;
    }
  }

  &-caption {
    font-size: 18px;
    color: black;
    font-weight: 600;

    .inactive & {
      color: $color-grey-dark;
    }
  }

  &-subcaption {
    color: $color-grey;
    font-size: 14px;
  }

  &:hover {
    background-color: $color-green-primary;
    border-color: $color-green-primary;

    > .Van-DashboardHeaderDropdownEntry-iconWrapper svg {
      fill: white;
    }

    > .Van-DashboardHeaderDropdownEntry-option {
      > .Van-DashboardHeaderDropdownEntry-subcaption {
        color: white;
      }

      > .Van-DashboardHeaderDropdownEntry-caption {
        color: white;
      }
    }
  }

  &.inactive {
    cursor: default;

    &:hover {
      background-color: $color-white;
      border-color: $color-grey-bright;

      > .Van-DashboardHeaderDropdownEntry-iconWrapper svg {
        color: $color-grey-nobel;
      }

      > .Van-DashboardHeaderDropdownEntry-option {
        > .Van-DashboardHeaderDropdownEntry-subcaption {
          color: $color-grey;
        }

        > .Van-DashboardHeaderDropdownEntry-caption {
          color: $color-grey-dark;
        }
      }
    }
  }
}

.Van-DashboardHeaderNotifications {
  height: 100%;
  position: relative;

  &:focus,
  &:active {
    outline: 0;
  }

  &--hasContent {
    > .Van-DashboardHeaderNotifications-toggle {
      cursor: pointer;
    }
  }

  &--active {
    background-color: $color-grey-bright;
  }

  svg {
    fill: $color-grey-nobel;
  }

  &-toggle {
    padding: 0 15px;
    height: 100%;
    display: flex;
    align-items: center;
  }

  &-count {
    $size: 15px;

    align-items: center;
    background-color: $color-blue-primary;
    border-radius: 50%;
    box-shadow: 1px 1px 2px #666;
    color: white;
    display: flex;
    font-size: 10px;
    font-weight: 600;
    height: $size;
    justify-content: center;
    left: 12px;
    max-height: $size;
    max-width: $size;
    overflow: hidden;
    position: absolute;
    top: 20px;
    width: $size;
    z-index: 21;
  }

  &-listWrapper {
    background-color: transparent;
    position: absolute;
    top: 80%;
    max-width: calc(100vw - 40px);
    width: 320px;
    right: 10px;
  }

  &-title {
    font-weight: 600;
    margin-top: 0;
    padding-right: 25px;
  }

  &-notification {
    background-color: $color-blue-bright;
    border-radius: 5px;
    padding: 15px;
    color: white;
    font-weight: normal;
    margin-bottom: 15px;
    position: relative;
    z-index: 21;

    &Close {
      cursor: pointer;
      position: absolute;
      right: 15px;
      top: 15px;

      svg {
        fill: $color-facebook-blue;
      }
    }
  }
}
