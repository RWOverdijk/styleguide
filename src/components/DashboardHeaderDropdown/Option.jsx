import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from '../Icon';

class Option extends React.Component {
  handleClickAndClose = (ev, inactive) => {
    if (inactive) {
      return;
    }
    this.props.handleClick(ev);
    this.props.closeMenu();
  };

  render() {
    const { icon, caption, subcaption, inactive } = this.props;
    const className = {
      inactive
    };

    return (
      <div
        className={classnames('Van-DashboardHeaderDropdownEntry', className)}
        onClick={ev => this.handleClickAndClose(ev, inactive)}
      >
        {icon !== null && (
          <div className="Van-DashboardHeaderDropdownEntry-iconWrapper">
            <Icon source={icon} />
          </div>
        )}

        <div className="Van-DashboardHeaderDropdownEntry-option">
          <div className="Van-DashboardHeaderDropdownEntry-caption">{caption}</div>

          {subcaption && (
            <div className="Van-DashboardHeaderDropdownEntry-subcaption">{subcaption}</div>
          )}
        </div>
      </div>
    );
  }
}

Option.propTypes = {
  icon: PropTypes.string,
  caption: PropTypes.node.isRequired,
  subcaption: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
  closeMenu: PropTypes.func,
  inactive: PropTypes.bool
};

Option.defaultProps = {
  icon: null,
  subcaption: null,
  closeMenu: () => {},
  inactive: undefined
};

export default Option;
