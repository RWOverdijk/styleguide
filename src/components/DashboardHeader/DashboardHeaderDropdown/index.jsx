import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from '../../Icon';
import angleUp from '../../../assets/icons/fa/angle-up.svg';
import angleDown from '../../../assets/icons/fa/angle-down.svg';
import SectionHeader from './SectionHeader';
import Option from './Option';

class DashboardHeaderDropdown extends React.Component {
  static propTypes = {
    icon: PropTypes.string,
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    children: PropTypes.node,
  };

  static defaultProps = {
    subtitle: null,
    children: null,
    icon: null,
  };

  state = {
    isActive: false
  };

  toggleMenu = () => {
    this.setState({ isActive: !this.state.isActive });
  };

  closeMenu = () => {
    this.setState({ isActive: false });
  };

  preventDefault = ev => ev.preventDefault();

  render() {
    const { title, subtitle, icon, children } = this.props;
    const { isActive } = this.state;
    const classes = classnames('Van-DashboardHeaderDropdown', {
      'Van-DashboardHeaderDropdown--active': this.state.isActive
    });

    return (
      <div className={classes}>
        <div
          className="Van-DashboardHeaderDropdown-captionWrapper"
          onClick={this.toggleMenu}
          onBlur={this.closeMenu}
          tabIndex="0"
        >
          {icon && (
            <div className="Van-DashboardHeaderDropdown-iconWrapper">
              <Icon source={icon} />
            </div>
          )}

          <div className="Van-DashboardHeaderDropdown-caption">
            <div className="Van-DashboardHeaderDropdown-title">{title}</div>
            {subtitle && <div className="Van-DashboardHeaderDropdown-subtitle">{subtitle}</div>}
          </div>

          <div className="Van-DashboardHeaderDropdown-toggle">
            <Icon source={isActive ? angleUp : angleDown} />
          </div>
        </div>

        {children && isActive && (
          <div className="Van-DashboardHeaderDropdown-optionWrapper" onMouseDown={this.preventDefault}>
            {children}
          </div>)}
      </div>
    );
  }
}

DashboardHeaderDropdown.SectionHeader = SectionHeader;
DashboardHeaderDropdown.Option = Option;

export default DashboardHeaderDropdown;
