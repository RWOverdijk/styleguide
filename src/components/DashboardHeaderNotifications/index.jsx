/* eslint-disable jsx-a11y/no-noninteractive-tabindex */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../Icon';
import bell from '../../assets/icons/fa/bell.svg';
import Notification from './Notification';

class DashboardHeaderNotifications extends React.Component {
  static propTypes = {
    children: PropTypes.node,
    activeByDefault: PropTypes.bool
  };

  static defaultProps = {
    children: null,
    activeByDefault: false
  };

  constructor(props) {
    super(props);

    this.state = {
      isActive: props.activeByDefault
    };
  }

  toggle = () => {
    if (!this.props.children) {
      return;
    }

    this.setState({ isActive: !this.state.isActive });
  };

  hide = () => {
    if (!this.props.children) {
      return;
    }

    this.setState({ isActive: false });
  };

  render() {
    const { children } = this.props;
    const notifications = React.Children.count(children);
    const classes = classNames(
      'Van-DashboardHeaderNotifications',
      { 'Van-DashboardHeaderNotifications--active': this.state.isActive },
      { 'Van-DashboardHeaderNotifications--hasContent': notifications > 0 }
    );

    return (
      <div className={classes} onBlur={this.hide} tabIndex="0">
        <div className="Van-DashboardHeaderNotifications-toggle" onClick={this.toggle}>
          {notifications > 0 && (
            <div className="Van-DashboardHeaderNotifications-count">{notifications}</div>
          )}

          <Icon source={bell} />
        </div>

        {notifications > 0 &&
          this.state.isActive && (
            <div
              className="Van-DashboardHeaderNotifications-listWrapper"
              onMouseDown={ev => ev.preventDefault()}
            >
              {children}
            </div>
          )}
      </div>
    );
  }
}

DashboardHeaderNotifications.Notification = Notification;

export default DashboardHeaderNotifications;
