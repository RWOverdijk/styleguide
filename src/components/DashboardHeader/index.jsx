import React from 'react';
import PropTypes from 'prop-types';
import logo from '../../assets/icons/vandebron.svg';
import './styles.scss';

/* eslint-disable react/no-danger */
const DashboardHeader = props => (
  <header className="Van-DashboardHeader">
    <section className="Van-DashboardHeader-logoWrapper">
      <div className="Van-DashboardHeader-logoText">Mijn</div>
      <span className="Van-DashboardHeader-logo" dangerouslySetInnerHTML={{ __html: logo }} />
    </section>

    {props.children && <section className="Van-DashboardHeader-addons">{props.children}</section>}
  </header>
);

DashboardHeader.defaultProps = {
  children: null
};

DashboardHeader.propTypes = {
  children: PropTypes.node
};

export default DashboardHeader;
export { default as DashboardHeaderDropdown } from './DashboardHeaderDropdown';
export { default as DashboardHeaderNotifications } from './DashboardHeaderNotifications';
