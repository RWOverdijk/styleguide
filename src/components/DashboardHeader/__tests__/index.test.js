import React from 'react';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DashboardHeader from '..';
import DashboardHeaderDropdown from '../../DashboardHeaderDropdown';
import DashboardHeaderNotifications from '../../DashboardHeaderNotifications';
import checkIcon from '../../../../dist/icons/check.svg';
import treeIcon from '../../../../dist/icons/tree-ftic.svg';
import drinksIcon from '../../../../dist/icons/drinks.svg';
import gearIcon from '../../../../dist/icons/gear.svg';

configure({ adapter: new Adapter() });

describe('<DashboardHeader />', () => {
  it('renders the correct markup', () => {
    const tree = renderer.create(<DashboardHeader />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('correctly renders children', () => {
    const tree = renderer.create(
      <DashboardHeader>
        <DashboardHeaderDropdown title="One dropdown" icon={treeIcon}>
          <DashboardHeaderDropdown.Option caption="One level" handleClick={() => {}} />
          <DashboardHeaderDropdown.Option caption="Two levels" subcaption="With a subcaption" handleClick={() => {}} />
        </DashboardHeaderDropdown>

        <DashboardHeaderDropdown title="Two dropdowns" icon={drinksIcon}>
          <DashboardHeaderDropdown.SectionHeader title="This one has a header" />
          <DashboardHeaderDropdown.Option caption="Wow, icons!" subcaption="So cool" icon={gearIcon} handleClick={() => {}} />
        </DashboardHeaderDropdown>

        <DashboardHeaderNotifications>
          <DashboardHeaderNotifications.Notification title="You got notifications" close={() => {}}>
            <div>
              One line<br />
              And a second line
            </div>
          </DashboardHeaderNotifications.Notification>
        </DashboardHeaderNotifications>
      </DashboardHeader>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<DashboardHeaderDropdown.Option', () => {
  it('calls the callback `handleClick` and closes the menu when an option is clicked', () => {
    const handleClick = jest.fn();
    const closeMenu = jest.fn();
    const component = mount(
      <DashboardHeaderDropdown.Option handleClick={handleClick} closeMenu={closeMenu} caption="test" />
    );

    component.find('.Van-DashboardHeaderDropdownEntry').simulate('click');
    expect(handleClick.mock.calls.length).toBe(1);
    expect(closeMenu.mock.calls.length).toBe(1);
  });

  it('renders inactive dropdown icon', () => {
    const handleClick = jest.fn();
    const closeMenu = jest.fn();
    const component = mount(
      <DashboardHeaderDropdown.Option
        inactive={true}
        caption="Inactive item"
        subcaption="Inactive item with a subcaption"
        handleClick={() => {}}
      />
    );

    const inactiveDropdownItem = component.find('.Van-DashboardHeaderDropdownEntry');
    expect(inactiveDropdownItem.hasClass('inactive')).toBe(true);
    component.find('.Van-DashboardHeaderDropdownEntry').simulate('click');
    expect(handleClick.mock.calls.length).toBe(0);
    expect(closeMenu.mock.calls.length).toBe(0);
  });
});

describe('<DashboardHeaderNotifications.Notification', () => {
  it('calls the callback `close` when the close icon is clicked', () => {
    const mock = jest.fn();
    const component = mount(
      <DashboardHeaderNotifications.Notification title="title" close={mock} children="" />
    );

    component.find('.Van-DashboardHeaderNotifications-notificationClose').simulate('click');
    expect(mock.mock.calls.length).toBe(1);
  });
});
