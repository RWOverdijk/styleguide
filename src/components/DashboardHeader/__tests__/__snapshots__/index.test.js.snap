// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardHeader /> correctly renders children 1`] = `
<header
  className="Van-DashboardHeader"
>
  <section
    className="Van-DashboardHeader-logoWrapper"
  >
    <div
      className="Van-DashboardHeader-logoText"
    >
      Mijn
    </div>
    <span
      className="Van-DashboardHeader-logo"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test-file-stub",
        }
      }
    />
  </section>
  <section
    className="Van-DashboardHeader-addons"
  >
    <div
      className="Van-DashboardHeaderDropdown"
    >
      <div
        className="Van-DashboardHeaderDropdown-captionWrapper"
        onBlur={[Function]}
        onClick={[Function]}
        tabIndex="0"
      >
        <div
          className="Van-DashboardHeaderDropdown-iconWrapper"
        />
        <div
          className="Van-DashboardHeaderDropdown-caption"
        >
          <div
            className="Van-DashboardHeaderDropdown-title"
          >
            One dropdown
          </div>
        </div>
        <div
          className="Van-DashboardHeaderDropdown-toggle"
        />
      </div>
    </div>
    <div
      className="Van-DashboardHeaderDropdown"
    >
      <div
        className="Van-DashboardHeaderDropdown-captionWrapper"
        onBlur={[Function]}
        onClick={[Function]}
        tabIndex="0"
      >
        <div
          className="Van-DashboardHeaderDropdown-iconWrapper"
        />
        <div
          className="Van-DashboardHeaderDropdown-caption"
        >
          <div
            className="Van-DashboardHeaderDropdown-title"
          >
            Two dropdowns
          </div>
        </div>
        <div
          className="Van-DashboardHeaderDropdown-toggle"
        />
      </div>
    </div>
    <div
      className="Van-DashboardHeaderNotifications Van-DashboardHeaderNotifications--hasContent"
      onBlur={[Function]}
      tabIndex="0"
    >
      <div
        className="Van-DashboardHeaderNotifications-toggle"
        onClick={[Function]}
      >
        <div
          className="Van-DashboardHeaderNotifications-count"
        >
          1
        </div>
      </div>
    </div>
  </section>
</header>
`;

exports[`<DashboardHeader /> renders the correct markup 1`] = `
<header
  className="Van-DashboardHeader"
>
  <section
    className="Van-DashboardHeader-logoWrapper"
  >
    <div
      className="Van-DashboardHeader-logoText"
    >
      Mijn
    </div>
    <span
      className="Van-DashboardHeader-logo"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test-file-stub",
        }
      }
    />
  </section>
</header>
`;
